echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/notification/MicrosoftTeamsNotificationAgent.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2020 Cerner Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.echo.notification;
+
+import com.netflix.spinnaker.echo.api.events.Event;
+import com.netflix.spinnaker.echo.microsoftteams.MicrosoftTeamsMessage;
+import com.netflix.spinnaker.echo.microsoftteams.MicrosoftTeamsService;
+import com.netflix.spinnaker.echo.microsoftteams.api.MicrosoftTeamsSection;
+import java.util.Map;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.text.WordUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Service;
+import retrofit.client.Response;
+import retrofit.mime.TypedByteArray;
+
+@Slf4j
+@ConditionalOnProperty("microsoftteams.enabled")
+@Service
+public class MicrosoftTeamsNotificationAgent extends AbstractEventNotificationAgent {
+
+  @Autowired private MicrosoftTeamsService teamsService;
```

Comment:
Use constructors instead of property injection.

Reply:
Converted to autowired constructor